serial/aio.py

```diff
@@ -1,121 +1,127 @@
-#!/usr/bin/env python3
-#
-# Experimental implementation of asyncio support.
-#
-# This file is part of pySerial. https://github.com/pyserial/pyserial
-# (C) 2015 Chris Liechti <cliechti@gmx.net>
-#
-# SPDX-License-Identifier:    BSD-3-Clause
-"""\
-Support asyncio with serial ports. EXPERIMENTAL
-
-Posix platforms only, Python 3.4+ only.
-
-Windows event loops can not wait for serial ports with the current
-implementation. It should be possible to get that working though.
-"""
 import asyncio
+import os
 import serial
-import logging
 
 
-class SerialTransport(asyncio.Transport):
-    def __init__(self, loop, protocol, serial_instance):
+__all__ = ["AsyncSerial"]
+
+
+class AsyncSerialBase:
+    def __init__(self, *args, loop=None, **kwargs):
+        self.ser = serial.serial_for_url(*args, **kwargs)
+
+        if loop is None:
+            loop = asyncio.get_event_loop()
```

Comment:
It doesn't necessarily create a loop. https://docs.python.org/3/library/asyncio-eventloops.html#asyncio.AbstractEventLoopPolicy.get_event_loop